weatherTask/weatherman.py

```diff
@@ -1,151 +1,146 @@
+        avg_temp_min = temp_min / data_points
+        avg_humidity = humidity / data_points
+
+        return avg_temp_max, avg_temp_min, avg_humidity
+
+    def draw_bar_chart(self, chart_data, chart_type):
+        """method to get bar chart data"""
+        for index in chart_data:
+            date = index['PKT'].strftime('%d')
+            if chart_type == "double-line":
+                print(f"{date} {colored('+' * index['Max TemperatureC'], 'red' )}"
+                f" {index['Max TemperatureC']}C")
+                print(f"{date} {colored('+' * index['Min TemperatureC'], 'blue')}"
+                f" {index['Min TemperatureC']}C")
+
+            elif chart_type == "single-line":
+                print(f"{date} {colored('+' * index['Min TemperatureC'], 'blue')}"
+                f"{colored('+' * index['Max TemperatureC'], 'red')}"
+                f" {index['Min TemperatureC']}C - {index['Max TemperatureC']}C")
 
 def main():
     """Main function defines arguments and its use cases"""
     parser = argparse.ArgumentParser(description="Generate weather reports.")
-    parser.add_argument("-e", "--year", type=int,
-    help="Get highest, lowest temperature, and humidity for a given year")
-    parser.add_argument("-a", "--avg", metavar=("year/month"),
-    help="Get average temperature and humidity for a given month")
-    parser.add_argument("-c", "--chart", metavar=("year/month"),
-    help="Draw a horizontal bar chart for a given month")
-    parser.add_argument("-cb", "--chartbar", metavar=("year/month"),
-    help="Draw a horizontal bar chart for a given month")
+    parser.add_argument("-e", "--year", type=int)
+    parser.add_argument("-a", "--avg", metavar="year/month")
+    parser.add_argument("-c", "--chart", metavar="year/month")
+    parser.add_argument("-cb", "--chartbar", metavar="year/month")
     parser.add_argument("path", type=str, help="Path to the weather data folder")
 
     args = parser.parse_args()
+    add_zero = '{:01d}'
 
     if args.year:
-        yearly_weather_data(args.year, args.path)
-        yearly_data = data
-        if yearly_data:
-            highest_temp, lowest_temp = highest_lowest_temperature(yearly_data)
-            highest_humid= highest_humidity(yearly_data)
-            print(f"Highest: {highest_temp['temp_max']}C on {highest_temp['date']}")
-            print(f"Lowest: {lowest_temp['temp_min']}C on {lowest_temp['date']}")
-            print(f"Humid: {highest_humid['humidity']}% on {highest_humid['date']}")
+        data = WeatherReport(args.year, args.path, months)
+        if data:
+            cal_data = data.data_filepath("yearly")
+            highest_temp, lowest_temp, highest_humid = data.high_low_temp(cal_data)
+            max_date = highest_temp['PKT'].strftime('%B %d')
+            low_date = lowest_temp['PKT'].strftime('%B %d')
+            high_date = highest_humid['PKT'].strftime('%B %d')
+            print(f"Highest: {highest_temp['Max TemperatureC']}C on {max_date}")
```

Comment:
Please create a separate class for handling all the printing, main function should just be receiving args and calling classes/functions